modules/integration/tests-integration/tests-es/src/test/java/org/wso2/carbon/registry/es/publisher/lifecycle/GRegPublisherLCStateDurationColorBoxTestCase.java

```diff
@@ -0,0 +1,281 @@
+import org.apache.wink.client.ClientResponse;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.testng.Assert;
+import org.testng.annotations.*;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.engine.frameworkutils.FrameworkPathUtil;
+import org.wso2.carbon.governance.lcm.stub.LifeCycleManagementServiceExceptionException;
+import org.wso2.carbon.integration.common.utils.exceptions.AutomationUtilException;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.registry.es.utils.GregESTestBaseTest;
+import org.wso2.carbon.registry.es.utils.LifeCycleBean;
+import org.wso2.carbon.registry.resource.stub.ResourceAdminServiceExceptionException;
+import org.wso2.greg.integration.common.clients.LifeCycleAdminServiceClient;
+import org.wso2.greg.integration.common.clients.LifeCycleManagementClient;
+import org.wso2.greg.integration.common.clients.ManageGenericArtifactAdminServiceClient;
+import org.wso2.greg.integration.common.clients.ResourceAdminServiceClient;
+import org.wso2.greg.integration.common.utils.GenericRestClient;
+import org.wso2.carbon.automation.test.utils.common.FileManager;
+import org.wso2.carbon.registry.es.utils.LifeCycleConstants;
+import org.wso2.carbon.registry.es.utils.LifeCycleUtils;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.core.MediaType;
+
+/***
+ * This class will test Greg Publisher Lifecycle State Duration and Its Color bellow the asset icon
+ */
+public class GRegPublisherLCStateDurationColorBoxTestCase extends GregESTestBaseTest {
+    public static final String CLASS_NAME = GRegPublisherLCStateDurationTestCase.class.getName();
+    private static final Log log = LogFactory.getLog(GRegPublisherLCStateDurationTestCase.class);
+    String assetId;
+    String cookieHeader;
+    GenericRestClient genericRestClient;
+    Map<String, String> queryParamMap;
+    Map<String, String> headerMap;
+    String publisherUrl;
+    ManageGenericArtifactAdminServiceClient manageGenericArtifactAdminServiceClient;
+    LifeCycleManagementClient lifeCycleAdminServiceClient;
+    ResourceAdminServiceClient resourceAdminServiceClient;
+    private LifeCycleManagementClient lifeCycleManagementClient;
+    private LifeCycleAdminServiceClient lifeCycleAdminService;
+
+    private TestUserMode userMode;
+    private ServerConfigurationManager serverConfigurationManager;
+
+    @Factory(dataProvider = "userModeProvider")
+    public GRegPublisherLCStateDurationColorBoxTestCase(
+            TestUserMode userMode) {
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    private static TestUserMode[][] userModeProvider() {
+        return new TestUserMode[][] { new TestUserMode[] { TestUserMode.SUPER_TENANT_ADMIN },
+                //                new TestUserMode[]{TestUserMode.TENANT_USER},
```

Comment:
Thank you Sameera for mentioning it. This code block has been refactored with new time in lifecycle tests